lib/node/runner/finish_ballot_test.go

```diff
@@ -0,0 +1,317 @@
+package runner
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	logging "github.com/inconshreveable/log15"
+	"github.com/stellar/go/keypair"
+	"github.com/stretchr/testify/require"
+
+	"boscoin.io/sebak/lib/ballot"
+	"boscoin.io/sebak/lib/block"
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/consensus"
+	"boscoin.io/sebak/lib/consensus/round"
+	"boscoin.io/sebak/lib/network"
+	"boscoin.io/sebak/lib/node"
+	"boscoin.io/sebak/lib/storage"
+	"boscoin.io/sebak/lib/transaction"
+	"boscoin.io/sebak/lib/transaction/operation"
+)
+
+func createNodeRunnerForTestingWithFileStorage(n int, conf common.Config, recv chan struct{}) (*NodeRunner, []*node.LocalNode, string) {
+	var ns []*network.MemoryNetwork
+	var net *network.MemoryNetwork
+	var nodes []*node.LocalNode
+	for i := 0; i < n; i++ {
+		_, s, v := network.CreateMemoryNetwork(net)
+		net = s
+		ns = append(ns, s)
+		nodes = append(nodes, v)
+	}
+
+	for j := 0; j < n; j++ {
+		nodes[0].AddValidators(nodes[j].ConvertToValidator())
+	}
+
+	localNode := nodes[0]
+	policy, _ := consensus.NewDefaultVotingThresholdPolicy(67)
+
+	connectionManager := NewTestConnectionManager(
+		localNode,
+		ns[0],
+		policy,
+		recv,
+	)
+
+	st := &storage.LevelDBBackend{}
+	dir, err := ioutil.TempDir("", "sebak-test")
+	if err != nil {
+		panic(err)
+	}
+
+	{
+		var err error
+		config, _ := storage.NewConfigFromString(fmt.Sprintf("file://%s", dir))
```

Comment:
We should really not be doing this in unittests. Can we move it to an integration test ?

Reply:
@Geod24 Hmmm, what is the problem of this? After testing, the temporary directory is removed.